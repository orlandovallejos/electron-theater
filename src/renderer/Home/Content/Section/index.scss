.wrapper {
  background-color: #101010;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}
