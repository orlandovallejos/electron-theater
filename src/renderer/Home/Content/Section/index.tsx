/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
// Components
import PosterItem from '../../../common/PosterItem';
// Types
import { MovieItem } from '../../../../types';
// Styles
import SectionWrapper from './style';

type Props = {
  items: MovieItem[];
};

const Section = (props: Props) => {
  const { items } = props;
  return (
    <SectionWrapper>
      {items.map((item) => {
        return (
          <PosterItem
            key={item.id}
            url={item.posterPath}
            title={item.title}
            voteAverage={item.voteAverage}
            releaseDate={item.releaseDate}
          />
        );
      })}
    </SectionWrapper>
  );
};

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default Section;
