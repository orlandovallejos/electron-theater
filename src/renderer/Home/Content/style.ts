/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import styled from 'styled-components';

const ContentWrapper = styled.div`
  background: #101010;
  padding: 30px 0;

  h1 {
    margin-left: 40px;
    color: #fff;
    font-size: 14px;
    text-transform: uppercase;
  }
`;
ContentWrapper.displayName = 'ContentWrapper';

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default ContentWrapper;
