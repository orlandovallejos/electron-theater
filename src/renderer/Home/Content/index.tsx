/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
import Section from './Section';
// Types
import { Movie } from '../../../types';
// Api
import moviesApi from '../../../api/Movies';
// Styles
import styles from './index.scss';

const Content = () => {
  const [movies, setMovies] = React.useState<Movie[]>([]);

  React.useEffect(() => {
    async function getMovies() {
      const data = await moviesApi.getTopMovies();
      setMovies(data.results);
    }
    getMovies();
  }, []);

  return (
    <div className={styles.wrapper}>
      <Section movies={movies} />
    </div>
  );
};

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default Content;
