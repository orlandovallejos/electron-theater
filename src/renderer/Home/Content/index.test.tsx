/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
import shallow from 'enzyme/shallow';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
import Content from './index';

describe('<Content />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Content movies={[]} series={[]} />);

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
