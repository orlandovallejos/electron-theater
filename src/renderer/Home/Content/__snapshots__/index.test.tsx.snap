// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render correctly 1`] = `
"<ContentWrapper>
  <h1>
    Trending movies
  </h1>
  <Section items={{...}} />
  <h1>
    Trending TV Shows
  </h1>
  <Section items={{...}} />
</ContentWrapper>"
`;
